diff.go

```diff
@@ -188,31 +203,43 @@ type groupDiff struct {
 	membersToRemove []YtsaurusMembership
 }
 
+type existedGroup struct {
+	ytsaurusGroup YtsaurusGroupWithMembers
+	sourceGroup   SourceGroup
+}
+
 func (a *App) diffGroups(
-	azureGroups []AzureGroupWithMembers,
+	sourceGroups []SourceGroupWithMembers,
 	ytGroups []YtsaurusGroupWithMembers,
-	usersMap map[AzureID]YtsaurusUser,
-) groupDiff {
+	usersMap map[ObjectID]YtsaurusUser,
+) (*groupDiff, error) {
 	var groupsToCreate, groupsToRemove []YtsaurusGroup
 	var groupsToUpdate []UpdatedYtsaurusGroup
 	var membersToAdd, membersToRemove []YtsaurusMembership
 
-	azureGroupsWithMembersMap := make(map[AzureID]AzureGroupWithMembers)
-	ytGroupsWithMembersMap := make(map[AzureID]YtsaurusGroupWithMembers)
-
-	for _, group := range azureGroups {
-		azureGroupsWithMembersMap[group.AzureID] = group
+	sourceGroupsWithMembersMap := make(map[ObjectID]SourceGroupWithMembers)
+	for _, group := range sourceGroups {
+		sourceGroupsWithMembersMap[group.SourceGroup.GetID()] = group
 	}
+
+	existedGroupsWithMembersMap := make(map[ObjectID]existedGroup)
 	for _, group := range ytGroups {
-		ytGroupsWithMembersMap[group.AzureID] = group
+		sourceGroup, err := a.buildSourceGroup(&group)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create azure group from source")
+		}
+		existedGroupsWithMembersMap[sourceGroup.GetID()] = existedGroup{group, sourceGroup}
```

Comment:
so existedGroup.sourceGroup is not used therefore code can be little simplified here as in diffUsers i suppose